src/sqlfluff/core/rules/base.py

```diff
@@ -802,28 +802,33 @@ def _choose_anchor_segment(
 
         anchor: BaseSegment = segment
         child: BaseSegment = segment
-        for seg in context.parent_stack[0].path_to(segment)[1:-1][::-1]:
-            # Which lists of children to check against.
-            children_lists: List[List[BaseSegment]] = []
-            if filter_meta:
-                # Optionally check against filtered (non-meta only) children.
-                children_lists.append(
-                    [child for child in seg.segments if not child.is_meta]
-                )
-            # Always check against the full set of children.
-            children_lists.append(seg.segments)
-            children: List[BaseSegment]
-            for children in children_lists:
-                if edit_type == "create_before" and children[0] is child:
-                    anchor = seg
-                    assert anchor.raw.startswith(segment.raw)
-                    child = seg
-                    break
-                elif edit_type == "create_after" and children[-1] is child:
-                    anchor = seg
-                    assert anchor.raw.endswith(segment.raw)
-                    child = seg
-                    break
+        if (
+            context.parent_stack[0]
+            and context.parent_stack[0].path_to(segment)
```

Comment:
I think we should assign this result to a variable that we reuse later. As this code is written, we'll call `path_to()` **three times**, and it's not a cheap function call. (It walks down the parse tree, potentially a long way.)

Reply:
Done.